### PR TITLE
feat(options): Add support for managed storage

### PR DESCRIPTION
**Pull request by smalluban**
Experimental support for the `ManagedStorage` native manifest.

On macOS create a file in `/Library/Application Support/Mozilla/ManagedStorage/firefox@ghostery.com.json`:

```json
{
  "name": "firefox@ghostery.com",
  "description": "",
  "type": "storage",
  "data": {
    "disableOnboarding": true,
    "disableUserControl": true,
    "trustedDomains": ["bbc.com"]
  }
}
```

More details about the file in other platforms: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_manifests#manifest_location

The managed options are created in the form of actions, which reflect changes in the main options:
* `disableOnboarding` - skips the onboarding, as it is approved by the administrator
* `disableUserControl` - disables every UI, which allows changing options - pausing, selective blocking, panel menu with links to settings, disabling preview trackers, manual access to the settings page and onboarding page (it redirects to `ghostery.com`)
* `trustedDomains` - an array of domains added permanently to the `options.paused`

### CODE REVIEW

File: src/pages/onboarding/index.js [hunk 1]
```diff
@@ -13,25 +13,30 @@ import { mount, html, store, router } from 'hybrids';
...
 import './elements.js';
 
 import Main from './views/main.js';
 import Success from './views/success.js';
 
-store.resolve(Options).then(({ onboarding }) => {
+store.resolve(Options).then(({ onboarding, managed }) => {
+  if (managed) {
+    return window.location.replace(`https://www.${GHOSTERY_DOMAIN}`);
+  }
```
Comment (chrmod) [inline src/pages/onboarding/index.js:26]: not needed

File: src/store/managed-options.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+ */
+
+import { store } from 'hybrids';
+import { debugMode } from '/utils/debug';
+
+export default {
+  supported: false,
+
+  enableTerms: false,
+  disableUserSettings: false,
```
Comment (chrmod) [inline src/store/managed-options.js:39]: `disableSettings`

File: src/store/managed-options.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+ */
+
+import { store } from 'hybrids';
+import { debugMode } from '/utils/debug';
+
+export default {
+  supported: false,
+
+  enableTerms: false,
```
Comment (chrmod) [inline src/store/managed-options.js:38]: `disableOnboarding`

**Review by chrmod — CHANGES_REQUESTED** [review]
we need one more option to `disablePause` from the panel

File: src/store/options.js [hunk 4]
```diff
@@ -147,7 +151,9 @@ const Options = {
         Promise.resolve().then(() => sync(options, keys));
       }
 
-      return options;
+      return __PLATFORM__ === 'firefox' || __PLATFORM__ === 'chromium'
+        ? applyManagedOptions(options)
```
[the remark this answers is not in the document]
Comment (smalluban) [reply src/store/options.js:155]: No: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/managed#browser_compatibility

**Comment by smalluban** [comment]
The Chrome schema has been added. Here you have docs about configuration it in different platforms:
https://www.chromium.org/administrators/configuring-policy-for-extensions

**Comment by philipp-classen** [comment]
For Chromium on Linux (tested on Arch):

Create a file /etc/chromium/policies/managed/test_policy.json (assumption: extension ID = "gjdacaklflccploihbcnabmhffhcbljm")

```
{
  "3rdparty": {
    "extensions": {
      "gjdacaklflccploihbcnabmhffhcbljm": {
        "disableOnboarding": true,
        "disableUserControl": true,
        "trustedDomains": ["bbc.com"]
      }
     }
   }
 }
```

chrome://policy/ shows
policy

**Review by chrmod — APPROVED** [review]

**Pull request merged**